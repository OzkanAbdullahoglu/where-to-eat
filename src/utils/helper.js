
export const openHoursManage = (obj) => {
  let willOpen = '';
  let willClosed = '';
  let openUntil = '';
  const date = new Date();
  const today = date.getDay();
  const time = parseInt(
    `${JSON.stringify(date.getHours())}${JSON.stringify(date.getMinutes())}`, 10
  );

  if (obj === undefined || obj.open[today] === undefined) {
    return '';
  }
  const closeTimeInt = parseInt(obj.open[today].end, 10);
  const openTime = `${obj.open[today].start.slice(0, 2)}:${obj.open[today].start.slice(2)}`;
  const closeTime = `${obj.open[today].end.slice(0, 2)}:${obj.open[
    today
  ].end.slice(2)}`;
  if (closeTimeInt - time < 100) {
    willClosed = `Close soon ${closeTime}`;
  } else if (time > closeTimeInt || time < closeTimeInt) {
    willOpen = `Opens at ${openTime}`;
  } else {
    openUntil = `Open until ${closeTime}`;
  }
  return (
    willClosed || willOpen || openUntil
  );
};

export const getCategoriesTitle = (arr) => {
  const titleArr = arr.map((obj) => obj.title);
  const concatenateTitles = titleArr.reduce(
    (accTitles, singleTitle) => `${accTitles}, ${singleTitle}`
  );
  return concatenateTitles;
};
