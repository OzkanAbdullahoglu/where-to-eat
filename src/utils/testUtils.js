/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';


export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
}
;
