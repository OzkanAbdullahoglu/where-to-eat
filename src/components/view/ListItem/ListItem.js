/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';
import ItemRating from '../ItemRating/ItemRating';
import ItemDetails from '../ItemDetail/ItemDetails';
import { openHoursManage } from '../../../utils/helper';

const ListItem = ({
  name,
  itemRate,
  reviewCounts,
  itemCost,
  itemAddress,
  itemHours,
  imageURL,
  yelpURL,
  itemCategories,
}) => {
  const fallBackImgURL = 'https://via.placeholder.com/80x92.png/ffdede/304368?text=NO+PHOTO';
  return (
    <a href={yelpURL} target="_blank" rel="noopener noreferrer">
      <div className="cards-result">
        <div className="cards-result-content">
          <div className="cards-result-text-content">
            <div className="cards-result-header">
              <div className="cards-result-title-container">
                <h3 className="cards-result-title">
                  <span>{name}</span>
                </h3>
                <ItemRating rating={itemRate} reviewCount={reviewCounts} />
              </div>
            </div>
            <div className="cards-result-details-container">
              <ItemDetails
                cost={itemCost}
                address={itemAddress}
                categories={itemCategories}
              />
            </div>
            <div className="cards-result-opening-hours">
              {openHoursManage(itemHours[0])}
            </div>
          </div>

          <div className="cards-image-container">
            <div
              className="cards-result-image"
              style={{
                backgroundImage: `url(${imageURL}), url(${fallBackImgURL})`,
              }}
            ></div>
          </div>
        </div>
      </div>
      <div className="cards-divider-bottom-line"></div>
    </a>
  );
};

ListItem.propTypes = {
  itemRate: PropTypes.number,
  reviewCounts: PropTypes.number,
  name: PropTypes.string,
  itemCost: PropTypes.string,
  itemHours: PropTypes.array,
  yelpURL: PropTypes.string,
  imageURL: PropTypes.array,
  itemCategories: PropTypes.array,
  itemAddress: PropTypes.string,
};

export default ListItem;
