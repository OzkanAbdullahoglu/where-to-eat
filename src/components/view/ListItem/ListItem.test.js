/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListItem from './ListItem';
import { openHoursManage } from '../../../utils/helper';

describe('List Item Component', () => {
  let openHoursManageMocked = jest.fn();

  const props = {
    name: 'test-name',
    itemRate: 5,
    reviewCounts: 123,
    itemCost: '$',
    itemAddress: 'test address',
    itemHours: [
      {
        open: [
          { is_overnight: false, end: '2200', start: '1200', day: 0 },
          { is_overnight: false, end: '2200', start: '1200', day: 1 },
          { is_overnight: false, end: '2200', start: '1200', day: 2 },
          { is_overnight: false, end: '2200', start: '1200', day: 3 },
          { is_overnight: false, end: '2200', start: '1200', day: 4 },
          { is_overnight: false, end: '2200', start: '1200', day: 5 },
          { is_overnight: false, end: '2200', start: '1200', day: 6 },
        ],
      },
    ],
    imageURL: ['test url'],
    yelpURL: 'test yelp url',
    itemCategories: [
      { title: 'Test title' },
    ],
  };

  const listItemWrapper = shallow(<ListItem {...props} />);
  it('renders List Component', () => {
    expect(toJson(listItemWrapper)).toMatchSnapshot();
  });
  it('contains a Item Details Component', () => {
    expect(listItemWrapper.find('ItemDetails').exists()).toBe(true);
  });
  it('contains a Item Rating Component', () => {
    expect(listItemWrapper.find('ItemRating').exists()).toBe(true);
  });

  describe('List Item Component Passing Props Correctly', () => {
    it('passing `rating` props', () => {
      const ratingProp = listItemWrapper.find('ItemRating').prop('rating');
      expect(ratingProp).toEqual(props.itemRate);
    });
    it('passing `reviewCount` props', () => {
      const reviewCountProp = listItemWrapper.find('ItemRating').prop('reviewCount');
      expect(reviewCountProp).toEqual(props.reviewCounts);
    });
    it('passing `cost` props', () => {
      const costProp = listItemWrapper.find('ItemDetails').prop('cost');
      expect(costProp).toEqual(props.itemCost);
    });
    it('passing `address` props', () => {
      const addressProp = listItemWrapper.find('ItemDetails').prop('address');
      expect(addressProp).toEqual(props.itemAddress);
    });
    it('passing `categories` props', () => {
      const categoriesProp = listItemWrapper.find('ItemDetails').prop('categories');
      expect(categoriesProp).toEqual(props.itemCategories);
    });
  });
  describe('List Item Component Render Props Correctly', () => {
    it('renders `name` props', () => {
      const nameSpanProp = listItemWrapper.find('.cards-result-title').children().text();
      expect(nameSpanProp).toEqual(props.name);
    });
    it('renders `yelpURL` props', () => {
      const yelpURLHrefProp = listItemWrapper.find('a').prop('href');
      expect(yelpURLHrefProp).toEqual(props.yelpURL);
    });
    it('renders `itemHours` props', () => {
      openHoursManageMocked = openHoursManage;
      const itemHoursProp = listItemWrapper.find('.cards-result-opening-hours').text();
      expect(itemHoursProp).toEqual(openHoursManageMocked(props.itemHours[0]));
    });
    it('renders `imageURL` props', () => {
      openHoursManageMocked = openHoursManage;
      const imageURLProp = listItemWrapper.find('.cards-result-image').prop('style');
      expect(imageURLProp.backgroundImage).toContain(props.imageURL);
    });
  });
});
