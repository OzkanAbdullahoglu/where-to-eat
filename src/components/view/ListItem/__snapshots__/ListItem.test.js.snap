// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Component renders List Component 1`] = `
<a
  href="test yelp url"
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="cards-result"
  >
    <div
      className="cards-result-content"
    >
      <div
        className="cards-result-text-content"
      >
        <div
          className="cards-result-header"
        >
          <div
            className="cards-result-title-container"
          >
            <h3
              className="cards-result-title"
            >
              <span>
                test-name
              </span>
            </h3>
            <ItemRating
              rating={5}
              reviewCount={123}
            />
          </div>
        </div>
        <div
          className="cards-result-details-container"
        >
          <ItemDetails
            address="test address"
            categories={
              Array [
                Object {
                  "title": "Test title",
                },
              ]
            }
            cost="$"
          />
        </div>
        <div
          className="cards-result-opening-hours"
        >
          Opens at 12:00
        </div>
      </div>
      <div
        className="cards-image-container"
      >
        <div
          className="cards-result-image"
          style={
            Object {
              "backgroundImage": "url(test url), url(https://via.placeholder.com/80x92.png/ffdede/304368?text=NO+PHOTO)",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="cards-divider-bottom-line"
  />
</a>
`;
